src/main/scala/org/musigma/sbt/rat/SbtRatPlugin.scala

```diff
@@ -37,93 +39,163 @@ object SbtRatPlugin extends AutoPlugin {
   override def requires = JvmPlugin
 
   object autoImport {
-    // FIXME: this should probably be a task, not a configuration (e.g., packageBin)
-    val Audit = config("audit")
+    val ratCheck = taskKey[Unit]("Performs a release audit check")
 
-    val auditCheck = taskKey[Unit]("Performs a release audit check")
-    val auditReport = taskKey[AuditReport]("Generates a release audit report")
+    type AuditReport = ClaimStatistic
+    val ratReport = taskKey[AuditReport]("Generates a release audit report")
 
-    val addDefaultLicenseMatchers = settingKey[Boolean]("Adds the default RAT license matchers to allowedLicenses")
+    val ratAddDefaultLicenseMatchers = settingKey[Boolean]("Whether to add the default list of license matchers.")
 
-    type LicenseFamily = ILicenseFamily
-    val allowedLicenseFamilies = settingKey[Seq[LicenseFamily]]("Which licenses families to allow")
+    val ratLicenseFamilies = settingKey[Seq[String]]("Specifies the license families to accept.")
 
-    type HeaderMatcher = IHeaderMatcher
-    val allowedLicenseHeaders = settingKey[Seq[HeaderMatcher]]("Which licenses to allow in file headers")
+    val ratLicenses = settingKey[Seq[(String, String, String)]]("Extra license to match--each Seq item is a tuple of (category, name, license text)")
 
-    type AuditReport = ClaimStatistic
+    val ratExcludes = settingKey[Seq[File]]("Files to exlucde from audit checks, relative to the baseDirectory")
```

Comment:
This type of thing would normally be done by `excludes in ratReport` or `includes in ratReport`, though I'm not sure how to implement that. I'd prefer consistency with most other sbt plugins here.

Reply:
I looked into this and couldn't figure out how to do it in a good way.  In order for "excludes in ratReport" to work, it looks like the ``val excludes = `` setting would need to be defined in the autoImport object. But that means it is auto imported into the global namespace. And a setting like ``excludes`` seems pretty generic and should probably be namespaced somehow. So if we kept the "rat" prefix, the build.sbt config would like like ``ratExcludes in ratReport := ...`` which feels a little odd to me.

Maybe something like ``excludes in ratReport`` is standard if ``excludes`` is a an already defined and standard Sbt key, and ratReport would want to override it? The [Sbt Plugin Best Practices Page](https://www.scala-sbt.org/1.0/docs/Plugins-Best-Practices.html) does something like that with ``sources in obfuscate``, but sources is already defined, so it's slightly different than 'excludes'. Other settings that are specific to their Obfucsate example are not scoped and and just prefixed with ``obfuscate``. 

But that's just my guess by reading the documentation. I'm not familiar with other plugins that do something like this (or many plugins in general). If you know of any, point them out and I can take a look at how they do it.

Reply:
Looks like I mixed up sbt and Maven. I meant `excludeFilter` and `includeFilter` which are defined in `sbt.Keys`.

Reply:
Ah, ok. My only thing against excludeFilter is that it's a pain to use, especially if you want to exclude specific paths instead of just names (which is what we want in Daffodil). However, this patchset actually ends up creating a FileFilter based on the ratExcludes value, so maybe that can be a good default for excludeFilter in ratReport, and if someone wants something different then they can override it? I think that would be fairly straightforward to implement.

Reply:
A good default for excludeFilter makes sense to me. The user can override the filter by either replacing it or appending to it, so it seems consistent with sbt practices.

Reply:
Ok, pushed three new commits. The first is unrelated but something I had hanging around. It gives a cleaner output when ratCheck finds unapproved licenses (i.e. no meaningless stack trace). The second just moves the existing functionality into objects to follow the best practices--makes things cleaner and easier to support the last commit. The last commit adds support for includeFilter and excludeFilter, with reasonable defaults that can be overridden or appended to if needed.